falcon/app.py

```diff
@@ -936,6 +926,8 @@ def _handle_exception(self, req, resp, ex, params):
         """
         err_handler = self._find_error_handler(ex)
 
+        # NOTE(caselit): Reset body, data and media before calling the handler
+        resp.body = resp.data = resp.media = None
```

Comment:
Shouldn't we move these statements inside under the `if err_handler is not None`?
In which scenario is it beneficial to reset the body if we haven't found any error handler?

Reply:
Yes, there should be no difference, since if we do not have an err_handler falcon does not manage the response and instead makes the wsgi serve handle it.
That should only happen on exception that do not inherit from `Exception` now, like `KeyboardInterrupt` or `SystemExit`, since we add an handler for `Exception`. This case should be very rare, so it's probably the same to have it inside or outside the if.